### PR TITLE
4 add testing

### PR DESCRIPTION
**Pull request by enki211**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # Sterling
 
+## Getting set up
```
Comment (anthony-jclark) [inline README.md]: Glad you added these instructions. Thanks.

File: README.md [hunk 2]
```diff
@@ -8,27 +22,29 @@ Inspiration taken from [rust](https://doc.rust-lang.org/grammar.html).
 Identifiers can start with a-zA-z and include a-zA-z, -, _ , 0-9
 ```
 statement             : let_declaration | expression_statement | newline
-let_declaration       : "let" id '=' expression newline
```
Comment (anthony-jclark) [inline README.md]: Nice changes to your grammar overall. Have you considered checking it out using Antlr?

File: src/lexer.cpp [hunk 3]
```diff
@@ -34,31 +34,9 @@ Token Lexer::get_next_token() {
 
     // Check for whitespace and skip
     // Since the language uses newlines instead of something like a semicolon
-    // we need to check for and return them
+    // we need to check for and return them. TODO: Don't use recurrsion
```
Comment (anthony-jclark) [inline src/lexer.cpp]: Should be able to eliminate recursion by simply incrementing the index into your input string.

File: src/lexer.cpp [hunk 4]
```diff
@@ -34,31 +34,9 @@ Token Lexer::get_next_token() {
...
     if (isspace(curr)) {
         next_char();
-        if (curr == '\n') {
-            t.type = TOKEN_TYPES::NEWLINE;
-            t.value = "\n";
-
-            // Go to the next line and reset the column
-            line++;
```
Comment (anthony-jclark) [inline src/lexer.cpp]: Did you decide to get rid of line and column tracking?

File: test/test.cpp [hunk 5]
```diff
@@ -0,0 +1,96 @@
+#define CATCH_CONFIG_MAIN
```
Comment (anthony-jclark) [inline test/test.cpp]: Nice set of tests! How are you going to split your tests up in the future (lexer, parser, etc.)?

Comment (enki211) [reply test/test.cpp]: That is an excellent question, I had not given that much thought but I think separating the tests(lexer, parser, interpreter) into different files and having one file that imports and runs all of them sounds the most logical right now.

**Pull request merged**